tests/py/dynamo/distributed/test_nccl_ops.sh

```diff
@@ -0,0 +1,137 @@
+#!/bin/bash
+
+check_command() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+ensure_installed() {
+    local pkg="$1"
+    if ! check_command "$pkg"; then
+        echo "$pkg is not installed. Installing $pkg..."
+
+        # Determine if sudo is needed
+        if check_command sudo; then
+            SUDO="sudo"
+        else
+            SUDO=""
+        fi
+
+        # Detect OS and install accordingly
+        OS="$(uname -s)"
+        if [[ "$OS" == "Linux" ]]; then
+            if check_command apt-get; then
+                $SUDO apt-get update && $SUDO apt-get install -y "$pkg"
+            fi
+        else
+            echo "Unsupported OS: $OS. Please install $pkg manually."
+            exit 1
+        fi
+    else
+        echo "$pkg is already installed."
+    fi
+}
+
+ensure_mpi_installed() {
+    local pkg="$1"
+    if dpkg -l | grep -q "$pkg"; then
+        echo "$pkg is already installed."
+    else
+        echo "$pkg is not installed. Installing $pkg..."
+
+        # Determine if sudo is needed
+        if check_command sudo; then
+            SUDO="sudo"
+        else
+            SUDO=""
+        fi
+
+        # Detect OS and install accordingly
+        OS="$(uname -s)"
+        if [[ "$OS" == "Linux" ]]; then
+            if check_command apt-get; then
+                $SUDO apt-get update && $SUDO apt-get install -y "$pkg"
+            fi
+        else
+            echo "Unsupported OS: $OS. Please install $pkg manually."
+            exit 1
+        fi
+    fi
+}
+
+ensure_pytest_installed(){
+    if check_command pip; then
+        echo "pip is installed, installing pytest..."
+        pip install pytest
+    else
+        echo "pip is not installed. Please install pip first."
+        exit 1
+    fi
+}
+
+echo "Setting up the environment"
+
+OS="$(uname -s)"
+ARCH="$(uname -m)"
+PYTHON_VERSION="$(python3 -c 'import sys; print(f"cp{sys.version_info.major}{sys.version_info.minor}")')"
```

Comment:
Why does it matter what python version the system is using? Can we just use the same .so for all python versions?